flax/core/scope.py

```diff
@@ -837,7 +838,7 @@ def bind(variables: VariableDict,
 
 def apply(fn: Callable[..., Any],
           mutable: CollectionFilter = False,
-          flags: Optional[Dict] = None) -> Callable[..., Any]:
+          flags: Optional[FrozenDict] = None) -> Callable[..., Any]:
```

Comment:
We seem to accept both `FrozenDict` and `Dict`, probably the correct type is either `Union[Dict, FrozenDict]` or even `Mapping`.

Reply:
I think changing to `Mapping` is best